src/client/cli/cmd/info.cpp

```diff
@@ -24,6 +24,28 @@
 namespace mp = multipass;
 namespace cmd = multipass::cmd;
 
+namespace
+{
+std::vector<mp::InstanceSnapshotPair> add_instance_and_snapshot_names(const mp::ArgParser* parser)
+{
+    std::vector<mp::InstanceSnapshotPair> instance_snapshot_names;
+
+    for (const auto& arg : parser->positionalArguments())
+    {
+        const auto tokens = arg.split('.');
+
+        mp::InstanceSnapshotPair inst_snap_name;
+        inst_snap_name.set_instance_name(tokens[0].toStdString());
+        if (tokens.size() > 1)
+            inst_snap_name.set_snapshot_name(tokens[1].toStdString());
```

Comment:
Instead of using split and possibly losing some part of the string, I changed it to set the instance name to everything before the first occurrence of a `.` if it exists and the snapshot to everything after. Since `.` is an invalid character for instances and snapshots regardless the daemon will return no results found if more periods/leading/ending periods are entered.

Reply:
OK. I am usually an adept of fail-fast, but I guess we can leave it up to the daemon to decide what names are and are not acceptable.